### PR TITLE
First pass on implementing Bookings

### PR DESCRIPTION
**Pull request by tbscanlon**
Right now they go into the DB. They're not yet displayed anywhere however.

The form is implemented too but it's very basic

### CODE REVIEW

File: app/controllers/bookings.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class MakersBnb < Sinatra::Base
+  get '/bookings/new' do
+    erb :'/bookings/new'
+  end
+
+  post '/bookings' do
+    start = Date.parse(params[:start_date])
+    finish = Date.parse(params[:end_date])
```
Comment (corina) [inline app/controllers/bookings.rb]: maybe a test with the app/ look in db before using parse

File: app/views/bookings/new.erb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+<form class="" action="/bookings" method="post">
+  <label for="start_date">Check In:
+    <input type="text" name="start_date" placeholder="DD/MM/YYYY">
+  </label>
+
+  <label for="end_date">Check Out:
+    <input type="text" name="end_date" placeholder="DD/MM/YYYY">
+  </label>
+
+  <input type="text" name="space_id" value="" placeholder="space if">
```
Comment (corina) [inline app/views/bookings/new.erb]: placeholder should be space id?

File: spec/backend/features/booking_spaces_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+feature 'Booking a Space' do
+  scenario 'User can book a space' do
+    sign_up
+    create_space
+    visit '/bookings/new'
+    fill_in :space_id, with: '1'
+    fill_in :start_date, with: "21/7/2017"
+    fill_in :end_date, with: "22/7/2017"
+    click_button("Book")
+    expect(Booking.count).to eq 1
```
Comment (corina) [inline spec/backend/features/booking_spaces_spec.rb]: extra test

**Review by corina — CHANGES_REQUESTED** [review]
🍔

File: app/views/bookings/new.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+<form class="" action="/bookings" method="post">
+  <label for="start_date">Check In:
+    <input type="text" name="start_date" placeholder="DD/MM/YYYY">
+  </label>
+
+  <label for="end_date">Check Out:
+    <input type="text" name="end_date" placeholder="DD/MM/YYYY">
+  </label>
+
+  <input type="text" name="space_id" value="" placeholder="space id">
```
Comment (vhonebon1) [inline app/views/bookings/new.erb]: agreed this will be taken out

**Review by vhonebon1 — APPROVED** [review]
comment refers to line that will be removed at a later date

**Review by marysalemme — APPROVED** [review]

File: app/controllers/bookings.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    erb :'/bookings/new'
+  end
+
+  get '/bookings/show/:id' do
+    @booking = Booking.get(params[:id])
+    erb :'/bookings/show'
+  end
+
+  post '/bookings' do
+    booking_error("You need to be signed in to book a space") unless current_user
```
Comment (corina) [inline app/controllers/bookings.rb]: wrap the 3 errors in a method

**Review by corina — CHANGES_REQUESTED** [review]
GG!

**Pull request merged**